manifests/install.pp

```diff
@@ -2,10 +2,38 @@
 #
 # This class is called from puppet_agent for install.
 #
-class puppet_agent::install {
+# === Parameters
+#
+# [package_file_name]
+#   The puppet-agent package file name.
+#   (see puppet_agent::prepare::package_file_name)
+#
+class puppet_agent::install(
+  $package_file_name = undef,
+) {
   assert_private()
 
-  package { $::puppet_agent::package_name:
-    ensure => present,
+  if $::operatingsystem == 'SLES' and $::operatingsystemmajrelease == '10' {
+    if $puppet_agent::params::master_agent_version != $::aio_agent_version {
+
+      contain puppet_agent::install::remove_packages
+
+      exec { 'replace puppet.conf removed by package removal':
```

Comment:
So manually passing in `-U --oldpackage` didn't seem to blow up... @haus Is this a bad idea?

```
txq76jldrr6fd3h:~ # rpm -i --replacepkgs --oldpackage puppet-agent-1.2.5-1.sles10.x86_64.rpm
warning: puppet-agent-1.2.5-1.sles10.x86_64.rpm: Header V3 RSA/SHA1 signature: NOKEY, key ID 4bd6ec30
error: Failed dependencies:
	pe-puppet conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-ruby-rgen conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-puppet-enterprise-release < 4.0.0 conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-facter conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-hiera conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-mcollective conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-mcollective-common conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-augeas conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-ruby conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-rubygems conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-libyaml conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-libldap conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-ruby-ldap conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-ruby-stomp conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-rubygem-deep-merge conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-ruby-shadow conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-ruby-augeas conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-openssl conflicts with puppet-agent-1.2.5-1.sles10.x86_64
	pe-virt-what conflicts with puppet-agent-1.2.5-1.sles10.x86_64

```
```
txq76jldrr6fd3h:~ # rpm -i -U --replacepkgs --oldpackage puppet-agent-1.2.5-1.sles10.x86_64.rpm
warning: puppet-agent-1.2.5-1.sles10.x86_64.rpm: Header V3 RSA/SHA1 signature: NOKEY, key ID 4bd6ec30
warning: /etc/puppetlabs/puppet/puppet.conf created as /etc/puppetlabs/puppet/puppet.conf.rpmnew
insserv: script puppet: service puppet already provided!
insserv: script pe-mcollective: service mcollective already provided!
pe-puppet                 0:off  1:off  2:off  3:off  4:on   5:off  6:off
pe-mcollective            0:off  1:off  2:on   3:on   4:on   5:on   6:off
```
```
txq76jldrr6fd3h:~ # rpm -U -U --replacepkgs --oldpackage puppet-agent-1.2.5-1.sles10.x86_64.rpm
warning: puppet-agent-1.2.5-1.sles10.x86_64.rpm: Header V3 RSA/SHA1 signature: NOKEY, key ID 4bd6ec30
```